packages/worker/src/worker-options.ts

```diff
@@ -193,12 +203,72 @@ export interface WorkerOptions {
   maxTaskQueueActivitiesPerSecond?: number;
 
   /**
-   * Maximum number of Workflow tasks to execute concurrently.
-   * Adjust this to improve Worker resource consumption.
-   * @default 100
+   * Maximum number of Workflow Tasks to execute concurrently.
+   *
+   * In general, a Workflow Worker's performance is mostly network bound (due to communication latency with the
+   * Temporal server). Accepting multiple Workflow Tasks concurrently helps compensate for network latency, until the
+   * point where the Worker gets CPU bound.
+   *
+   * Increasing this number will have no impact if Workflow Task pollers can't fill available execution slots fast
+   * enough. Therefore, when adjusting this value, you may want to similarly adjust `maxConcurrentWorkflowTaskPolls`.
+   * See {@link WorkerOptions.maxConcurrentWorkflowTaskPolls} for more information.
+   *
+   * Also, setting this value too high might cause Workflow Task timeouts due to the fact that the Worker is not able
+   * to complete processing accepted Workflow Tasks fast enough. Increasing the number of Workflow threads
+   * (see {@link WorkerOptions.workflowThreadPoolSize}) may help in that case.
+   *
+   * As a general guidelines:
+   * - High latency to Temporal Server => Increase this number
+   * - Very short Workflow Tasks (no lengthy Local Activities) => increase this number
+   * - Very long/heavy Workflow Histories => decrease this number
+   * - Low CPU usage despite backlog of Workflow Tasks => increase this number
+   * - High number of Workflow Task timeouts => decrease this number
+   *
+   * In our reference performance test against Temporal Cloud, running with a single Workflow thread and the Reuse V8
+   * Context option enabled, we reached peak performance with a `maxConcurrentWorkflowTaskExecutions` of `120`, and
+   * `maxConcurrentWorkflowTaskPolls` of `60`. Your millage may vary.
```

Comment:
You might want to mention the latency from where you tested and the type of load if relevant